Framework/PythonInterface/plugins/algorithms/MaskShadowedPixels.py

```diff
@@ -0,0 +1,178 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#pylint: disable=no-init,invalid-name
+import mantid
+import numpy as np
+
+
+class MaskShadowedPixels(mantid.api.PythonAlgorithm):
+    def category(self):
+        return "DataHandling\\Nexus"
+
+    def seeAlso(self):
+        return ["MaskShadowedPixels"]
+
+    def name(self):
+        return "MaskShadowedPixels"
+
+    def summary(self):
+        return "Generate maks file for NOMAD to exclude shadowed pixels."
+
+    def PyInit(self):
+        self.declareProperty(mantid.api.WorkspaceProperty('InputWorkspace', '', mantid.kernel.Direction.Input), "Workspace to save")
+        self.declareProperty(mantid.api.FileProperty('OutputFilename', '', action=mantid.api.FileAction.Save, extensions=['.gss']),
+                             doc='Name of the GSAS file to save to')
+
+        # TODO - Add missing input properties
+
+    def validateInputs(self):
+        """Validate input properties
+        (virtual method)
+        """
+        issues = dict()
+
+        return issues
+
+    def PyExec(self):
+        """
+        Main method to execute the algorithm
+        """
+        # Prepare input workspace: normalize by solid angle
+        raw_workspace = self.getProperty('InputWorkspace').value
+        assert raw_workspace, 'Input workspace cannot be None'
+        intensity_array = self.prepare_input(raw_workspace)  # return (101378,) array
+
+        # Parse YMAL configuration
+        config = self.parse_yaml(self.getProperty('ConfigurationFile').value)
+        assert isinstance(config, dict)
+
+        # Get instrument information
+        instrument_info = self.set_nomad_constants()
+
+        # Initialize the empty mask list
+        mask_array = np.zeros(shape=(instrument_info.num_pixels,), dtype='float')
+        assert mask_array
+
+        # Determine thresholds for a given tube (part 1)
+        # TASK 335
+        self.determine_tube_threshold(intensity_array, instrument_info, config)
+        # END OF Part1 ---
+
+        # Calculate the median for a particular Group of banks (part 2)
+        # TASK 331
+        self.calculate_banks_medians(config)
+        # END OF Part2 ----
+
+        # Determine thresholds for a given bank (part 3)
+        # TASK 333
+        self.determine_banks_thredsholds(config)
+        # END OF Part3 ---
+
+        # Process for output
```

Comment:
I don't think this algorithm is complete.
Are all the changes pushed to remote?